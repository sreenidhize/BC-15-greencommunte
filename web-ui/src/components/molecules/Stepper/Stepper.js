import React from "react";
import StepperLabel from "../../atoms/StepperLabel/StepperLabel";
import LabelNumber from "../../atoms/StepperNumber/LabelNumber";

const Stepper = (props) => {
  return (
    <div style={{ display: "flex", alignItems: "center" }} data-testid="stepper" >
      <LabelNumber isVisited={props.isVisited} number={props.number} />
      <StepperLabel label={props.label} isVisited={props.isVisited} />
    </div>
  );
};

export default Stepper;
