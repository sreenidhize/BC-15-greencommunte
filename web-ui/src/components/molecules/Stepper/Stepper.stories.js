import React from 'react';
import { ThemeProvider } from '@material-ui/core';
import theme from '../../../utils/theme/theme';
import Stepper from './Stepper';

export default {
    title: 'molecules/Stepper',
    component: Stepper,
};

const Template = (args) => <ThemeProvider theme={theme}><Stepper {...args} /></ThemeProvider>;

export const Primary = Template.bind({});

Primary.args = {
    number:"1",  
    label: "Your Location", 
    isVisited: true,
};