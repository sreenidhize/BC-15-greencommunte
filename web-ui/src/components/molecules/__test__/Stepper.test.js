import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Stepper from "../Stepper/Stepper";


describe("Stepper test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <Stepper number="1"  label="Your Location" isVisited={false}/>
      );
      expect(container).toMatchSnapshot();
    });
  
    test("Test inactive Stepper  render", () => {
      const { getByTestId } = render(<Stepper number="1"  label="Your Location" isVisited={false}/>);
      expect(getByTestId("stepper")).toBeInTheDocument();
    });
  
    test("Test active Stepper render", () => {
      const { getByTestId } = render(<Stepper number="2"  label="Job Location" isVisited={true}/>);
      expect(getByTestId("stepper")).toBeInTheDocument();
    });
  });