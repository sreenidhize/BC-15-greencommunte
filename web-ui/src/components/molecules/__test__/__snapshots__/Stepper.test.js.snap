// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="stepper"
    style="display: flex; align-items: center;"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle makeStyles-unvisited-1 MuiAvatar-colorDefault"
      data-testid="label-number"
    >
      1
    </div>
    <p
      class="MuiTypography-root makeStyles-root-3 MuiTypography-body1"
      data-testid="stepper-label"
      style="color: rgb(95, 115, 129); font-size: 18px; margin-right: 15px;"
    >
      Your Location
    </p>
  </div>
</div>
`;
